service/index/writer.go

```diff
@@ -334,28 +340,23 @@ func (w *Writer) Close() error {
 func (w *Writer) flush() {
 	defer w.wg.Done()
 
+	ticker := time.NewTicker(w.cfg.FlushInterval)
```

Comment:
You could have just deferred `ticker.Stop()` and kept the `return` in the for loop, no?